tests/tt_metal/tt_metal/perf_microbenchmark/8_dram_adjacent_core_read/test_dram_read.cpp

```diff
@@ -512,6 +515,192 @@ void get_dram_reader_core_coords_wormhole_b0(
     all_cores_ordered = adj_core_logical_realloc;
 }
 
+void get_dram_reader_core_coords_blackhole(
+    tt_metal::Device* device, CoreRangeSet& all_cores, std::vector<CoreCoord>& all_cores_ordered) {
+    // hardcoded for bh
+    uint32_t full_grid_size_x = 17;
+    uint32_t full_grid_size_y = 12;
```

Comment:
Can these values be taken from `device->grid_size()` instead of hardcoded here?
Also seems like WH and BH functions for `get_dram_reader_core_coords_*` are basically identical apart from hardcoding these values. Would it be a good idea to commonize it to reduce duplication?

Reply:
yes it should be commonized and be uplifted to understand harvested columns. just copied and pasted to run the test